markdown/core/src/main/kotlin/org/jetbrains/jewel/markdown/rendering/DefaultMarkdownBlockRenderer.kt

```diff
@@ -127,83 +122,18 @@ public open class DefaultMarkdownBlockRenderer(
 
     @Composable
     override fun render(block: Heading, styling: MarkdownStyling.Heading) {
-        when (block) {
-            is H1 -> render(block, styling)
-            is H2 -> render(block, styling)
-            is H3 -> render(block, styling)
-            is H4 -> render(block, styling)
-            is H5 -> render(block, styling)
-            is H6 -> render(block, styling)
+        when (block.level) {
+            1 -> render(block, styling.h1)
+            2 -> render(block, styling.h2)
+            3 -> render(block, styling.h3)
+            4 -> render(block, styling.h4)
+            5 -> render(block, styling.h5)
+            6 -> render(block, styling.h6)
         }
```

Comment:
Missing the else case:
```suggestion
            6 -> render(block, styling.h6)
            else -> error("Heading level ${block.level} not supported:\n$block")
        }
```